apps/openmw/mwrender/sky.cpp

```diff
@@ -1337,6 +1340,87 @@ class RainFader : public AlphaFader
     osg::Uniform* mRainIntensityUniform;
 };
 
+void SkyManager::setCamera(osg::Camera *camera)
+{
+    mCamera = camera;
+}
+
+class WrapAroundOperator : public osgParticle::Operator
+{
+public:
+    WrapAroundOperator(osg::Camera *camera, osg::Vec3 wrapRange): osgParticle::Operator()
```

Comment:
To avoid a redundant copy, this variable can be passed by const reference:

`..., const osg::Vec3& wrapRange)`

I usually don't care about performance issues in places where it's not important at all, but this particular one is something cppcheck (a static analyser) will complain about so I prefer to get them fixed everywhere and keep the code consistent.

The same applies to using 'i++' instead of '++i' because the former (in theory) creates an extra temporary variable (according to cppcheck, though I highly doubt that modern compilers don't already optimize this in all cases).

Reply:
Okay, I will get cppcheck too so I can also see these things.

Reply:
Should I pre-increment everywhere in the code then?

Reply:
> Should I pre-increment everywhere in the code then?

Sure.

Admittedly, It's been a while since I last cppchecked. If it no longer flags those things, feel free to keep the code as is - but it still might be nice to for the sake of consistency.

Reply:
(That was the last nitpick, I promise. Will test now and merge when I come back)

Reply:
Cppcheck doesn't complain but I will change to pre-increment anyway. It found some memory leaks in other files BTW. I'm OK with this nitpicking, I'm learning a lot.

Reply:
We also have Coverity Scan, but it's not been run for quite a while. Updated the branch and:

>  834 files changed, 27339 insertions(+), 9312 deletions(-)

This should be interesting.

Reply:
Had to disable a bunch of stuff to get the build under the time limit, now Coverity Scan seems to have completed and found some new defects.

However, I can't actually view them as I seem to have locked myself out of the OpenMW project on Coverity when I changed my github email. @psi29a you have an account there right, can you let me in?